pkg/handler/collector/deps_dev/deps_dev.go

```diff
@@ -133,6 +135,22 @@ func (d *depsCollector) populatePurls(ctx context.Context, docChannel chan<- *pr
 	if err != nil {
 		return fmt.Errorf("unable to retrieve datasource: %w", err)
 	}
+
+	if !d.retrieveDependencies {
+		var wg sync.WaitGroup
+		wg.Add(len(ds.PurlDataSources))
+
+		for _, purl := range ds.PurlDataSources {
+			go func(p datasource.Source) {
+				defer wg.Done()
+				d.retrieveMetadataWithoutDeps(ctx, p, docChannel)
```

Comment:
Calling this `func` concurrently via `go func` will cause race conditions as the state is manipulated by the calling funcs. 

This needs a larger refactor than just calling concurrently.